### ISSUE TITLE
Add country filtering for score rankings

### ISSUE THREAD

**Issue by TPGPL** [ppy/osu-web#4816]
(Description added by peppy because OP was too lazy to even bother.)

This is a proposal to add (back) the ability to view country-specific score rankings. This was unintentionally available by clicking country ranking then clicking a country then clicking score ranking (the filter stayed in place) but was not accessible via any other means. Has been removed for now (#4812) to fix serious performance degradation.

If to be implemented, we will need new database indices on all related tables.

**Comment by peppy** [comment]
Please add a description stating what this is and why you want it.

**Comment by celerih** [comment]
As PP is being abused a lot of players have found enjoyment in instead focusing on score rankings. I myself know a few people personally who relied on those leaderboards to compare themselves and see their progress especially before they can join the global leaderboard. It helps to offer an alternative way to play and rank up in the game as it was earlier and a lot of people still get enjoyment out of it.

**Comment by WalterToro** [comment]
Personally, even if people like this feature, there are some things that should be addressed/fixed first, for example, the treatment of unrestricted players, as they dont get their scores wiped after an unrestriction, so they can get double the score if they pass the same songs again, after that i dont see issues on bringing this back if people want alternative ways to compete.

**Comment by Natsukiii11** [comment]
i agree with what @WaterToro said completely. there are definitely a few things that should be addressed but other than that, there isnt a reason to not have it, as far as i can see

**Comment by JBHyperion** [comment]
I followed my country score leaderboards semi-regularly since (for catch at least), the active UK community is quite small and the top <x> players haven't really changed in an eternity. Whilst I'm not saying score ranking is by any means "better" than the PP system, it did offer a different dynamic that (for our community at least), enabled more competition.

I'd love to see this back if a less convoluted / resource-intensive solution can be found, but if we can take this as an opportunity to fix more glaring issues like the above first though, then I completely agree they should be prioritised.

**Comment by Remyria** [comment]
I think it's a nice "little" feature that deserves to exist.

**Comment by Yentis** [comment]
I feel if the overhead was large enough to notice then that just goes to show that this feature was being used a lot, regardless of the reason. If we always keep tunnel vision on the "big fish" issues then no quality of life changes will ever happen. 

You can already see this in the feature request forums, countless topics with a crazy amount of support but nonetheless they don't get added because there's always something more important.

**Comment by peppy** [comment]
The issue was caused by one user spidering the pages using automation, not continued use. In fact, the fact we *didn't* notice the issue for over a year means it was *not* being used. I confirmed this via metrics before deciding on forward action.

**Comment by RyShemycal** [comment]
I know that this ranking has many problems that must be fixed, but this game has the potential to have many more rankings apart from the main ones such as Perfomance points or Global Ranked Score, it would also serve to motivate other players to reach a more goal beyond the classic world ranking.

**Comment by FreddyBadAdd** [comment]
I agree with person above and want the feature added as much, if not more than others (:
Was a really nice way to track my score grind progress back when i was below 10b

**Comment by insomnyawolf** [comment]
So will this be a thing again at some point or is just dead?

**Comment by MrWither83** [comment]
With how much the Score Farming community has grown in the past year, I think this issue deserves more attention. 
There's still no way for all the beginners to know their ranked score rank, and comparing yourself to other score players in your country is really tedious.

**Comment by 0xallie** [comment]
I'm confused, how is this different from the existing per-beatmap "Country Ranking" available for supporters? Or was it re-added but this issue wasn't closed?

**Comment by insomnyawolf** [comment]
> I'm confused, how is this different from the existing per-beatmap "Country Ranking" available for supporters? Or was it re-added but this issue wasn't closed?

It is the same but for Profile score not just for each beatmap.
Something that lets you see the global score ranking.
